File: src/main/java/com/bot/tasks/LeaveGuildDeferredTask.java
package com.bot.tasks;

import com.bot.db.MembershipDAO;
import com.bot.utils.Logger;
import net.dv8tion.jda.core.entities.Member;
import net.dv8tion.jda.core.events.guild.GuildLeaveEvent;

public class LeaveGuildDeferredTask extends Thread {

    private final MembershipDAO membershipDAO;
    private GuildLeaveEvent guildLeaveEvent;
    private Logger logger;

    public LeaveGuildDeferredTask(GuildLeaveEvent guildLeaveEvent) {
        this.guildLeaveEvent = guildLeaveEvent;
        this.membershipDAO = MembershipDAO.getInstance();
        this.logger = new Logger(this.getClass().getName());
    }

    @Override
    public void run() {
        for (Member m : guildLeaveEvent.getGuild().getMembers()) {
            membershipDAO.removeUserMembershipToGuild(m.getUser().getId(), guildLeaveEvent.getGuild().getId());
        }

        logger.info("Left guild: " + guildLeaveEvent.getGuild().getName() + " with " + guildLeaveEvent.getGuild().getMembers().size() + " members");
    }
}

File: src/main/java/com/bot/tasks/LeaveGuildDeferredTask.kt
package com.bot.tasks

import com.bot.db.GuildDAO
import com.bot.db.MembershipDAO
import com.bot.db.RssDAO
import com.bot.db.ScheduledCommandDAO
import com.bot.utils.Logger
import net.dv8tion.jda.api.events.guild.GuildLeaveEvent

class LeaveGuildDeferredTask(private val guildLeaveEvent: GuildLeaveEvent) : Thread() {

    private val membershipDAO: MembershipDAO = MembershipDAO.getInstance()
    private val guildDAO: GuildDAO = GuildDAO.getInstance()
    private val scheduledDAO: ScheduledCommandDAO = ScheduledCommandDAO.getInstance()
    private val rssDAO = RssDAO.getInstance()
    private val logger: Logger = Logger(this.javaClass.name)

    override fun run() {
        for (m in guildLeaveEvent.guild.members) {
            membershipDAO.removeUserMembershipToGuild(m.user.id, guildLeaveEvent.guild.id)
        }

        guildDAO.setGuildActive(guildLeaveEvent.guild.id, false)
        scheduledDAO.removeAllScheduledInGuild(guildLeaveEvent.guild.id)
        rssDAO.removeAllSubsInGuild(guildLeaveEvent.guild.id)

        logger.info("Left guild: " + guildLeaveEvent.guild.name + " with " + guildLeaveEvent.guild.members.size + " members")
    }
}
